packages/traceability/issuer-irec-api/src/pods/certificate/handler/get-irec-certificates-to-import.handler.ts

```diff
@@ -17,20 +18,29 @@ export class GetIrecCertificatesToImportHandler
         private readonly queryBus: QueryBus,
         @Inject(IREC_SERVICE)
         private readonly irecService: IrecService,
-        private readonly deviceRegistryService: DeviceRegistryService
+        private readonly deviceRegistryService: DeviceRegistryService,
+        private readonly irecDeviceService: DeviceService
     ) {}
 
     async execute({ user }: GetIrecCertificatesToImportCommand): Promise<IrecAccountItemDto[]> {
         const irecCertificates = await this.irecService.getCertificates(user);
         const devices = await this.deviceRegistryService.find({ where: { owner: user.ownerId } });
+        const irecDevices = await this.irecDeviceService.findAll({
+            where: { ownerId: user.ownerId }
+        });
 
         const certificationRequests = await this.queryBus.execute<
             GetAllCertificationRequestsQuery,
             FullCertificationRequestDTO[]
         >(new GetAllCertificationRequestsQuery({ deviceIds: devices.map((d) => d.id) }));
 
-        return irecCertificates.filter((issue) => {
-            return !certificationRequests.some((cr) => issue.asset === cr.irecAssetId);
-        });
+        return irecCertificates
+            .filter((issue) => {
+                return !certificationRequests.some((cr) => issue.asset === cr.irecAssetId);
+            })
+            .map((issue) => {
+                const irecDevice = irecDevices.find((d) => d.code === issue.device.code);
```

Comment:
if you use `some` instead of `find` you already receive boolean, therefore no need to convert to Boolean (with !! which I don't like) later on

Reply:
fixed